repos/spack_repo/builtin/build_systems/compiler.py

```
@@ -219,7 +219,7 @@ def _fortran_path(self) -> Optional[str]:
 
 @memoized
 def _compiler_output(
-    compiler_path: Path, *, version_argument: str, ignore_errors: Tuple[int, ...] = ()
+    compiler_path: Path, *, version_argument: Optional[str], ignore_errors: Tuple[int, ...] = ()
```

A couple things:

* Looking at the body of this, there appears to be a sort-of optional mechanism already because it checks `if not version_argument` and then invokes `compiler()` without it if so (I could see an argument made for refactoring that, but for now, IMO we might as well follow what this allows for)
* (question) given prior point, and that you set the version arg to "" for windows, I would think this wouldn't have required overriding `determine_version`, but did an error occur when you didn't?

---

I find passing `""` as an alias for `None` and relying on unadvertised, subtle behavior for that to be a valid check to be fragile/error prone/hard to read/reason about vs being explicit that the given value is optional. 

For Windows we need to ignore errors from the compiler invocation. The intel and MSVC compilers both have no version argument, the version is determined by invoking the compiler with `LOGO` enabled (default behavior without `/NOLOGO`). The best way to do so is to just invoke the compiler with no arguments, but that returns an error code since the compiler complains about no provided source file.